### ISSUE TITLE
Utils:  Investigate our use of the orderedset library and remove it if possible, otherwise replace and update it

### ISSUE THREAD

**Issue by ccostino** [GSA/notifications-utils#103]
The utils project pulls in the [`orderedset`](https://pypi.org/project/orderedset/) library and makes use of it in several spots:

- notifications_utils/field.py
- notifications_utils/insensitive_dict.py
- notifications_utils/markdown.py
- notifications_utils/recipients.py
- notifications_utils/template_change.py
- setup.py
- tests/test_recipient_csv.py
- tests/test_template_types.py

This library hasn't been updated in some time and specifically mentions only working with Python versions 2.7 - 3.8.  It appears to work with Python 3.9 as well, but we know it does not work with 3.11+, which will prevent us from upgrading Python in the future.

Our preference would be to remove the library entirely if we don't need it.  If we do need it, then we'll have to research some alternatives; [a list of ordered set packages can be found on PyPI](https://pypi.org/search/?q=orderedset).

## Why is this an external library?

Because an `OrderedSet` doesn't exist in the standard library as of yet.  See [this discussion on the Python Ideas list](https://discuss.python.org/t/add-orderedset-to-stdlib/12730) for more details.

## Implementation Sketch and Acceptance Criteria

- [ ] Verify whether or not we need to use `orderedset` in the utils module, or if there's another way to implement the code with the desired effect; if we can do this another way without the use of an external library, then remove it entirely.
- [ ] If we can't remove it, then we need to find a suitable replacement
   - [ ] Check to see if the API and Admin can make do with leveraging anything to do with an `OrderedSet` from just utils itself.
   - [ ] Research some alternative packages ([a list can be found on PyPI](https://pypi.org/search/?q=orderedset) to start with), paying close attention to the following:
      - Does it do what we need it to do?
      - Does it work with recent versions of Python?
      - Does it require a different set of paradigms, or is it a drop in replacement for the use of `set` in Python?
      - Is it well maintained and has it been updated recently?
      - Is the author known, and/or is there a support community?
   - [ ] Make the swap in packages within the `pyproject.toml` file, update the `poetry.lock` file, and update the code and tests as necessary.

## Security Considerations

- We'll need to make sure the project is well-maintained and also adheres to security best practices.
- Removing or updating this library will enable to us to keep our own apps well-maintained and update other dependencies or Python itself as needed.

**Comment by terrazoon** [comment]
1. I tried replacing orderedset with ordered-set-37 in notifications-utils.  This was easy to do and all the utils tests passed.  Unfortunately, when I tried to run notifications-admin with this version of notifications-utils, there were many mysterious test failures.   Even though ordered-set-37 is supposed to be backward compatible, I'm not totally sure it is.

2. I then tried replacing OrderedSet with list, and just writing some extra code to prevent duplicates and so on.  This turned out to be very problematical because OrderedSet is used a lot in the templating code, and the developers were found of the '|' operator.   It would be a substantial effort to go down this road (5 days?).
